.github/workflows/build_and_push_to_docker.yml

```diff
@@ -30,23 +28,128 @@ jobs:
         password: ${{ secrets.DOCKER_PASSWORD }}
 
     - uses: actions/checkout@v2
-    - name: Build renku project docker images
+    - name: Build renku project python+ docker images
```

Comment:
Shouldn't this job be split into two steps? First building the python base images and then the images that depend on them? Right now the cuda images are not built correctly at all. It should be something like

```
jobs:
  base-python:
    <build base python using the renku-version matrix>
  base-extensions:
    depends-on: base-python
    <build extensions, passing the correct python base image using `--build-arg` to `docker build`>
```

Reply:
Yeah I couldn't figure that out -- tripped up by the matrix.

I'll try out your suggestion, thanks!

Reply:
in terms of building the cuda images correctly and being able to specify a latest as default that relies on the right image for the cuda images, I think we should also move to the naming convention already in the makefiles so that this makes sense: `renku/renkulab-py3.7:latest`

and then we also have:
`renku/renkulab-bioc:3_10-renku0.10.2-hash`
`renku/renkulab-r:3.6.3-renku0.10.2-hash`
...
etc.

Reply:
i didn't make the naming changes for now, just pointed at the current latest for default (and passed in the bulid-arg so the default will never be used anyway in the action)